akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRecoveringSnapshot.scala

```diff
@@ -79,87 +68,61 @@ private[akka] class EventsourcedRecoveringSnapshot[C, E, S](
     event match {
       case Some(evt) ⇒
         setup.log.error(cause, "Exception in receiveRecover when replaying snapshot [{}]", evt.getClass.getName)
-        Behaviors.stopped
-
-      case None ⇒
+      case _ ⇒
         setup.log.error(cause, "Persistence failure when replaying snapshot")
-        Behaviors.stopped
     }
+
+    Behaviors.stopped
   }
 
   private def onRecoveryTick(snapshot: Boolean): Behavior[InternalProtocol] =
     if (snapshot) {
       // we know we're in snapshotting mode; snapshot recovery timeout arrived
       val ex = new RecoveryTimedOut(s"Recovery timed out, didn't get snapshot within ${setup.settings.recoveryEventTimeout}")
-      onRecoveryFailure(ex, event = None)
+      onRecoveryFailure(ex, None)
     } else same // ignore, since we received the snapshot already
 
-  // protect against snapshot stalling forever because of journal overloaded and such
-  private val RecoveryTickTimerKey = "recovery-tick"
-
-  private def startRecoveryTimer(): Unit = {
-    setup.timers.startPeriodicTimer(RecoveryTickTimerKey, RecoveryTickEvent(snapshot = false), setup.settings.recoveryEventTimeout)
-  }
-
-  private def cancelRecoveryTimer(timers: TimerScheduler[_]): Unit = timers.cancel(RecoveryTickTimerKey)
-
   def onCommand(cmd: IncomingCommand[C]): Behavior[InternalProtocol] = {
     // during recovery, stash all incoming commands
-    setup.internalStash.stash(cmd) // TODO move stash out as it's mutable
+    setup.internalStash.stash(cmd)
     Behavior.same
   }
 
-  def onJournalResponse(response: JournalProtocol.Response): Behavior[InternalProtocol] = try {
-    throw new Exception("Should not talk to journal yet! But got: " + response)
-  } catch {
-    case NonFatal(cause) ⇒
-      returnRecoveryPermitOnlyOnFailure(cause)
-      throw cause
+  def onJournalResponse(response: JournalProtocol.Response): Behavior[InternalProtocol] = {
+    setup.log.debug("Unexpected response from journal: [{}], may be due to an actor restart, ignoring...", response)
```

Comment:
Not sure we dare `toString` on those messages, class name should be enough and safe.

Reply:
and yes, people use debug level in production

Reply:
good, agreed -- just getName here